packages/javascript-mutator/src/JavaScriptMutator.ts

```diff
@@ -8,14 +8,33 @@ import copy from './helpers/copy';
 import { NODE_MUTATORS_TOKEN, NodeMutator } from './mutators/NodeMutator';
 
 export class JavaScriptMutator implements Mutator {
-  public static inject = tokens(commonTokens.logger, NODE_MUTATORS_TOKEN);
-  constructor(private readonly log: Logger, private readonly mutators: readonly NodeMutator[]) {}
+  public static inject = tokens(commonTokens.logger, commonTokens.mutatorDescriptor, NODE_MUTATORS_TOKEN);
+  constructor(
+    private readonly log: Logger,
+    private readonly mutatorDescriptor: MutatorDescriptor,
+    private readonly mutators: readonly NodeMutator[]
+  ) {
+    /*
+      Edge case check
+      Check if `mutatorDescriptor` has been passed.
+      If it is not and `mutators` isn't as well, `mutatorDescriptor` is undefined
+      If it is not, but `mutators` is, `mutatorDescriptor` is an array
+    */
+    if (typeof mutatorDescriptor !== 'object' || Array.isArray(mutatorDescriptor)) {
+      this.mutatorDescriptor = {
+        babelPlugins: [],
+        excludedMutations: [],
+        name: ''
+      };
+    }
```

Comment:
This constructor implementation can be removed. We can rely on the type safety of TypeScript. When we receive a `MutatorDescriptor` here, we don't need to validate it again.